operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventProcessor.java

```diff
@@ -376,6 +365,8 @@ public synchronized void stop() {
 
   @Override
   public void start() throws OperatorException {
+    // on restart new executor service is created and needs to be set here
+    executor = ExecutorServiceManager.instance().executorService();
```

Comment:
Shouldn't the executor service be shut down then in `stop`?

Reply:
it is: https://github.com/java-operator-sdk/java-operator-sdk/blob/b29500f9674f4ec512540cd0a371901d4319908b/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ExecutorServiceManager.java#L100-L109